src/order/conditionally_complete_lattice.lean

```diff
@@ -769,3 +769,182 @@ instance (α : Type*) [conditionally_complete_linear_order α] :
   ..order_dual.decidable_linear_order α }
 
 end order_dual
+
+section with_top_bot
+
+open lattice
+
+open_locale classical
+/-- Extends `Sup` from a conditionally complete lattice `α` to `with_top α`.
+The new `Sup` returns a non-junk value for more subsets (e.g. for unbounded-above subsets of `α`). -/
+noncomputable instance with_top.conditionally_complete_lattice.has_Sup
+  {α : Type*} [conditionally_complete_lattice α] : has_Sup (with_top α) :=
+⟨λ S, if ⊤ ∈ S then ⊤ else
+      let So := (coe ⁻¹' S : set α) in
+      if bdd_above So then ↑(Sup So) else ⊤⟩
+
+/-- Extends `Inf` from a conditionally complete lattice `α` to `with_top α`. -/
+noncomputable instance with_top.conditionally_complete_lattice.has_Inf
+  {α : Type*} [conditionally_complete_lattice α] : has_Inf (with_top α) :=
+⟨λ S, if S ⊆ {⊤} then ⊤ else ↑(Inf (coe ⁻¹' S) : α)⟩
+
+/-- Adding a top element to a conditionally complete lattice gives a conditionally complete lattice -/
+noncomputable instance with_top.conditionally_complete_lattice
+  {α : Type*} [conditionally_complete_lattice α] :
+  conditionally_complete_lattice (with_top α) :=
+{ le_cSup := λ S a hS haS,
+    let So := (coe ⁻¹' S : set α) in
+    show a ≤ (ite (⊤ ∈ S) ⊤ (ite (bdd_above So) ↑(Sup So) ⊤)),
+    begin
+      split_ifs,
+      { exact le_top},
+      { cases a, contradiction, exact with_top.some_le_some.2 (le_cSup h_1 haS)},
+      { exact le_top},
+    end,
+  cSup_le := λ S a hS haS,
+    let So := (coe ⁻¹' S : set α) in
+    show (ite (⊤ ∈ S) ⊤ (ite (bdd_above So) ↑(Sup So) ⊤)) ≤ a,
+    begin
+      split_ifs,
+      { exact haS h},
+      { cases a, exact le_top,
+        refine with_top.some_le_some.2 (cSup_le _ _),
+        { rcases (set.exists_mem_of_ne_empty hS) with ⟨⟨⟩|b, hb⟩, contradiction,
+          rw set.ne_empty_iff_exists_mem, exact ⟨b, hb⟩},
+        { intros b hb, exact with_top.some_le_some.1 (haS hb)}},
+      { cases a, exact _root_.le_refl _,
+        exfalso, apply h_1, use a, intros b hb, exact with_top.some_le_some.1 (haS hb)},
+    end,
+  cInf_le := λ S a hS haS,
+    show (ite (S ⊆ {⊤}) ⊤ ↑(Inf (coe ⁻¹' S))) ≤ a,
+    begin
+      cases a, exact le_top,
+      split_ifs,
+      { rcases (h haS) with ⟨⟨⟩⟩ | _ | _},
+      { refine with_top.some_le_some.2 (cInf_le _ haS),
+        rcases hS with ⟨⟨⟩|a, ha⟩,
+        { use a, intros b hb, exact false.elim (with_top.not_top_le_coe b (ha hb))},
+        { use a, intros b hb, exact with_top.some_le_some.1 (ha hb)}}
+  end,
```

Comment:
the indentation of `end` should match that of `begin`, here and below.

Reply:
They are definitely now more consistent than they were.